app/controllers/wiki_controller.rb

```diff
@@ -72,6 +72,10 @@ def edit
     else
       @node = DrupalNode.find_wiki(params[:id])
     end
+    if @node.has_power_tag('locked') && (current_user.role != "admin" && current_user.role != "moderator")
```

Comment:
Ah! Here's the problem! `has_power_tag` looks for a tag with the given key, and any value, in the format `key:value` -- but the `locked` tag is not a `key:value` format tag. Here we should instead just use `@node.has_tag('locked')`. That should fix it!